healthchain/pipeline/basepipeline.py

```
@@ -0,0 +1,463 @@
+            self._components.append(new_component)
+        elif position in ["after", "before"]:
+            if not reference:
+                raise ValueError(
+                    f"Reference must be provided for position '{position}'."
+                )
+            offset = 1 if position == "after" else 0
+            self._insert_relative_position(new_component, reference, offset)
+        else:
+            raise ValueError(
+                f"Invalid position '{position}'. Must be 'first', 'last', 'after', 'before', or 'default'."
+            )
+
+    def _insert_relative_position(self, component, reference, offset):
+        """
+        Insert a component relative to a reference component in the pipeline.
+
+        Args:
+            component (PipelineNode): The component to be inserted.
+            reference (str): The name of the reference component.
+            offset (int): The offset from the reference component (0 for before, 1 for after).
+
+        Raises:
+            ValueError: If the reference component is not found in the pipeline.
+        """
+        ref_index = next(
+            (i for i, c in enumerate(self._components) if c.name == reference), None
+        )
+        if ref_index is None:
+            raise ValueError(f"Reference component '{reference}' not found.")
+
+        self._components.insert(ref_index + offset, component)
+
+    def remove(self, component_name: str) -> None:
+        """
+        Removes a component from the pipeline.
+
+        Parameters:
+        - component_name (str): The name of the component to be removed.
+
+        Returns:
+        None
+
+        Logs:
+        - WARNING: If the component is not found in the pipeline or fails to be removed.
+        """
+        # Check if the component exists in the pipeline
+        if not any(c.name == component_name for c in self._components):
+            logger.warning(f"Component '{component_name}' not found in the pipeline.")
```

No real issue here but I'm just wondering whether this edge case should fail louder? 

Specifically I see this warning being triggered by typos on r some underlying mistake. I think the user in that case would want to be explicitly thrown an error to help them catch the mistake.

---

good point